### DISCUSSION TITLE
LGBMClassifier.fit() got an unexpected keyword argument 'early_stopping_rounds'

### DISCUSSION THREAD

**Discussion by skancharapu775** [Diyago/Tabular-data-generation#72] [Q&A]
Hello,

When I run the GANGenerator function from the below file as well as for my own use-case, I get the error: "LGBMClassifier.fit() got an unexpected keyword argument 'early_stopping_rounds'". I did not get the error previously when running the code, and the old version of my code did not appear to work.
(link: https://colab.research.google.com/github/jeffheaton/t81_558_deep_learning/blob/master/t81_558_class_07_5_tabular_synthetic.ipynb#scrollTo=L-i4CdwYkgLU)

Another tabgan user asked it on chegg at this link: 
https://www.chegg.com/homework-help/questions-and-answers/traying-run-code-link-https-colabresearchgooglecom-github-jeffheaton-t81-558-deep-learning-q118173829),

The Chegg answerer said: **To fix the error message you are encountering, you need to update the LightGBM library version used in the code. The 'early_stopping_rounds' parameter was introduced in a later version of LightGBM, which is not compatible with the version currently installed.**

The following below is the steps given by the Chegg answerer to solve the problem that may warrant changes to the adversarial model file:
----------------------------------------------------------------------------------------------------------------------
To fix the error message you are encountering, you need to update the LightGBM library version used in the code. The 'early_stopping_rounds' parameter was introduced in a later version of LightGBM, which is not compatible with the version currently installed.

To resolve this issue, follow the steps below:

**Step 1**: Install the required libraries Before running the code, make sure you have the necessary libraries installed. In this case, you need to install the 'tabgan' library. Open a terminal or command prompt and run the following command:

```
pip install tabgan
```

**Step 2:** Update LightGBM You need to update the LightGBM library to a version that supports the 'early_stopping_rounds' parameter. To do this, modify the code as follows:

1. Import the lightgbm library:

```
import lightgbm as lgb
```

2.Update the LightGBM classifier initialization:

```
clf = lgb.LGBMClassifier(
    boosting_type='gbdt',
    objective='binary',
    num_leaves=31,
    learning_rate=0.05,
    n_estimators=1000,
    feature_fraction=0.8,
    bagging_fraction=0.8,
    bagging_freq=5,
    verbose=-1
)
```

**Step 2/2**
3.Remove the 'early_stopping_rounds' parameter from the fit() method

```
clf.fit(X_train, y_train, eval_set=[(X_valid, y_valid)], eval_metric='auc', verbose=200)
```

Step 3: Run the code After making the necessary updates, you can run the code again. The error message should no longer appear, and you should be able to obtain the desired result ('gen_x').

Explanation:
Note: It's always a good practice to check for library version compatibility when using different code sources, as libraries and their parameters may change over time.

Final answer
In short, the approach to solve this problem is to:
- Install a newer version of LightGBM.
- Run the code again.
- The error should be gone and you should be able to get the result gen_x.
---------------------------------------------------------------------------------------------------------------------- 

Please let me know if this is a problem on my end or these changes need to be implemented into source code.

I appreciate your help!
